cratedb_toolkit/datasets/model.py

```diff
@@ -0,0 +1,67 @@
+import dataclasses
+import typing as t
+
+import requests
+import sqlparse
+
+from cratedb_toolkit.util import DatabaseAdapter
+
+
+@dataclasses.dataclass
+class Dataset:
+    """
+    Wrapper around a dataset.
+    """
+
+    name: str
+    data_url: str
+    load_url: str
+
+    # Native datasets wrapper handle, e.g. `datasets`
+    # https://github.com/huggingface/datasets
+    handle: t.Optional[t.Any] = None
+
+    def to_cratedb(
+        self,
+        dburi: str,
+        table: str,
+        if_exists: t.Literal["append", "noop", "replace"] = "noop",
+        drop: bool = False,
+    ):
+        """
+        Load dataset into CrateDB, using SQL.
+        """
```

Comment:
Coming from https://github.com/crate/cratedb-examples/pull/352, there have been concerns about downloading and processing 200 MB worth of data again and again, so we figured this subsystem may need a resource saver.

FYI: This has been implemented here on behalf of the `if_exists=noop` default, which includes the complete shopping list of downloading the dataset only once, and importing it only once, regardless of how many times corresponding instructions are invoked.

In this spirit, it aims for [DWIM](https://en.wikipedia.org/wiki/DWIM), specifically in cases of one-shot data imports within presentation scenarios like this one. Sure enough, people will find use cases also beyond presentation purposes only. 🍀 

/cc @marijaselakovic, @karynzv, @surister, @hlcianfagna, @hammerhead, @ckurze